identity/src/main/java/com/stripe/android/identity/viewmodel/IdentityViewModel.kt

```diff
@@ -726,14 +736,26 @@ internal class IdentityViewModel constructor(
         APIException::class
     )
     suspend fun postVerificationPageData(
-        collectedDataParam: CollectedDataParam,
-        clearDataParam: ClearDataParam
-    ) =
+        collectedDataParam: CollectedDataParam
+    ): VerificationPageData {
         identityRepository.postVerificationPageData(
             verificationArgs.verificationSessionId,
             verificationArgs.ephemeralKeySecret,
             collectedDataParam,
-            clearDataParam
+            calculateClearDataParam(collectedDataParam)
+        ).let {
+            _collectedData.update { oldValue ->
```

Comment:
note: this is the place when page transition happens and the `VerificationPageData  POST` request to server is returned - after confirming server receives the updated data, we accumulate the newly collected data to `_collectedData`